app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java

```diff
@@ -0,0 +1,146 @@
+package com.example.teamyuml.decentworkmobile.views;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.example.teamyuml.decentworkmobile.R;
+import com.example.teamyuml.decentworkmobile.VolleyInstance;
+import com.example.teamyuml.decentworkmobile.utils.CreateJson;
+import com.example.teamyuml.decentworkmobile.utils.UserAuth;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Class responsible for adding engagments
+ * Parsing data to JSON and sending to the server
+ */
+public class EngagmentAdd extends AppCompatActivity implements View.OnClickListener {
+    private EditText advertisment_title_input;
```

Comment:
Why you call all of these `advetisment`? First this should be `advertisement`? I think announcement or notice are better words.